ipywidgets/widgets/interaction.py

```diff
@@ -217,6 +217,11 @@ def call_f(name=None, old=None, new=None):
         if manual:
             manual_button.disabled = True
         try:
+            def close_result_widgets(w):
+                if type(w.result) is Box:
+                    close_result_widgets(w.result)
+                    w.result.close()
+            close_result_widgets(container)
```

Comment:
Ahh, you're right. I had misread `interact()` and thought it was standard behavior to attach the result to the container, but that is not the case.  I agree this approach probably wouldn't do much for most people (and it would require some pretty ugly code to work at all for `interact()` rather than `interactive()`). 

(Also, I realized if `result` is just a plain `Box()`, the code currently generates an `AttributeError` when it tries to access `result`, but that's a more minor fix)